### ISSUE TITLE
Shell Command Agent Custom Event FR

### ISSUE THREAD

**Issue by phra** [cantino/huginn#1944]
**FEATURE REQUEST**

it would be nice to have the ability to pass arbitrary properties of the in event in the out event in order to implement a pipeline of shell command agents in which the previous agent pass along some of its parameters to the next agent.
what do you think about it?

**Comment by phra** [comment]
actually it's more general than #1868, because of it also permits to emit properties that don't exist in the input event.

**Comment by dsander** [comment]
> it also permits to emit properties that don't exist in the input event.

You can insert arbitrary properties to any Event with a `EventFormattingAgent`, an other option would be to include the data in the output of the command you are calling.

**Comment by phra** [comment]
@dsander i have two shell commands agent, i'm passing via manual event agent a configuration object that i want to propagate from one shell command agent to the next one. at the moment there is no way to do that and i do not want to write a _hack_ to work with the output of previous agent. @cantino

**Comment by dsander** [comment]
Yes that makes sense, but that's what has been requested in #1868 as well. I did not understand for which use case your addition is required.
